ros_cross_compile/ros_cross_compile.py

```diff
@@ -174,7 +184,15 @@ def cross_compile_pipeline(
 def main():
     """Start the cross-compilation workflow."""
     args = parse_args(sys.argv[1:])
-    cross_compile_pipeline(args)
+    data_collector = DataCollector()
+
+    with DataTimer() as end_end_timer:
+        cross_compile_pipeline(args, data_collector)
+
+    end_end_time = Datum('End_To_End-time', end_end_timer.elapsed,
+                         'seconds', str(datetime.now()))
```

Comment:
avoid hardcoding strings

Reply:
or if enum aren't possible, just having a named constant could be OK `SECONDS = 'seconds'`